### PR TITLE
[Logger] Unexpected keyNotFound error #420

### LINKED ISSUE #420: keyNotFound
**Describe the bug**
After the socket link succeeds, console Logs displays information about decode failures

**SDK Version**
 - Client: v0.9.3-rc.0
 - Version v0.9.3-rc.0

**To Reproduce**
1. Click on Scan QR code
2. After the socket connected , the log displays information about decode failures

**Screenshots**
WX20220809-101719

**Device (please complete the following information):**
 - Device: iPhone7
 - OS: iOS15.0.2

### PR DESCRIPTION
**Pull request by flypaper0**
# Description

- Unexpected keyNotFound error removed
- Serializer refactor
- Logging improvements

Resolves #420

## How Has This Been Tested?
Perform steps from #420 manually 

## Due Dilligence

* [ ] Breaking change
* [ ] Requires a documentation update

### CODE REVIEW

File: Sources/Chat/NetworkingInteractor.swift [hunk 1]
```diff
@@ -86,7 +86,7 @@ class NetworkingInteractor: NetworkInteracting {
         } else if let deserializedJsonRpcError: JSONRPCErrorResponse = serializer.tryDeserialize(topic: topic, encodedEnvelope: encodedEnvelope) {
             handleJsonRpcErrorResponse(response: deserializedJsonRpcError)
         } else {
-            print("Warning: Networking Interactor - Received unknown object type from networking relay")
+            logger.error("Warning: Networking Interactor - Received unknown object type from networking relay")
```
Comment (llbartekll) [inline Sources/Chat/NetworkingInteractor.swift:89]: I would not use logger.error here
maybe logger.warning?

File: Sources/WalletConnectRelay/Dispatching.swift [hunk 2]
```diff
@@ -82,9 +82,9 @@ final class Dispatcher: NSObject, Dispatching {
 
     private func dequeuePendingTextFrames() {
         while let frame = textFramesQueue.dequeue() {
-            send(frame) { error in
+            send(frame) { [weak self] error in
```
Comment (llbartekll) [inline Sources/WalletConnectRelay/Dispatching.swift:85]: would `unowned self` be better?

**Review by llbartekll — APPROVED** [review]
👍

File: Sources/WalletConnectRelay/Dispatching.swift (hunk 2 above)
Comment (flypaper0) [reply Sources/WalletConnectRelay/Dispatching.swift:85]: replaced weak with unowned everywhere in Dispatcher. Should work as we have Relay singleton

**Pull request merged**